lm_eval/models/huggingface.py

```diff
@@ -168,13 +171,17 @@ def __init__(
                 max_cpu_memory,
                 offload_folder,
             )
-        self.model = self._create_auto_model(
-            pretrained=pretrained,
-            revision=revision,
-            subfolder=subfolder,
-            torch_dtype=_get_dtype(dtype, self._config),
-            **accelerate_kwargs,
-        )
+
+        if is_random: 
+            self.model = self._create_random_model(config=self._config)
+        else:
+            self.model = self._create_auto_model(
+                pretrained=pretrained,
+                revision=revision,
+                subfolder=subfolder,
+                torch_dtype=_get_dtype(dtype, self._config),
+                **accelerate_kwargs,
```

Comment:
Hi, yes I noticed that too. From the [docs](https://huggingface.co/transformers/v3.0.2/model_doc/auto.html) it looks like AutoModel.from_config() can only take config as input so we don't have the option to pass the same parameters as AutoModel.from_pretrained(). However, use_accelerate defaults to False anyway so I am not currently using it in either case. I agree that when we use much larger models we might need to find a way to handle the random model but maybe this is ok for now and we can cross that bridge if/when we come to it?

Reply:
actually, having thought about this some more I think it's not an issue. There probably is no point loading a very large random model because it will have the exact same performance as a very small random model, so we can just stick with what we have for getting the random base line.

Reply:
oh ye good point 😂